src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataSetStep.test.tsx

```diff
@@ -0,0 +1,582 @@
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import noop from 'lodash/noop';
+import React from 'react';
+import { PublicationReleaseSummary } from '@common/services/publicationService';
+
+describe('DataSetStep', () => {
+  const testSubjects: Subject[] = [
+    {
+      id: 'subject-1',
+      name: 'Subject 1',
+      content: '<p>Test content 1</p>',
+      timePeriods: {
+        from: '2018/19',
+        to: '2020/21',
+      },
+      geographicLevels: ['Local Authority District', 'Ward'],
+      file: {
+        id: 'file-1',
+        name: 'Subject 1',
+        fileName: 'file-1.csv',
+        extension: 'csv',
+        size: '10 Mb',
+        type: 'Data',
+      },
+      filters: ['School type'],
+      indicators: ['Headcount', 'Percent'],
+    },
+    {
+      id: 'subject-2',
+      name: 'Subject 2',
+      content: '<p>Test content 2</p>',
+      timePeriods: {
+        from: '2015',
+        to: '2020',
+      },
+      geographicLevels: ['National', 'Local Authority'],
+      file: {
+        id: 'file-2',
+        name: 'Subject 2',
+        fileName: 'file-2.csv',
+        extension: 'csv',
+        size: '20 Mb',
+        type: 'Data',
+      },
+      filters: ['Ethnicity', 'FSM'],
+      indicators: ['Authorised absence rate'],
+    },
+  ];
+
+  const testFeaturedTables: FeaturedTable[] = [
+    {
+      id: 'featured-1',
+      name: 'Test featured 1',
+      description: 'Test featured description 1',
+      subjectId: 'subject-1',
+    },
+    {
+      id: 'featured-2',
+      name: 'Test featured 2',
+      description: 'Test featured description 2 find me',
+      subjectId: 'subject-1',
+    },
+    {
+      id: 'featured-3',
+      name: 'Test featured 3',
+      description: 'Test featured description 3',
+      subjectId: 'subject-3',
+    },
+  ];
+
+  const wizardProps: InjectedWizardProps = {
+    shouldScroll: true,
+    stepNumber: 1,
+    currentStep: 1,
+    isActive: true,
+    isEnabled: true,
+    isLoading: false,
+    setCurrentStep: (step, task) => task?.(),
+    goToNextStep: task => task?.(),
+    goToPreviousStep: task => task?.(),
+  };
+
+  const testRelease: Partial<PublicationReleaseSummary> = {
+    id: 'release-1',
+  };
+
+  test('renders radios with details if no `renderHighlightLink`', () => {
+    render(
+      <DataSetStep {...wizardProps} subjects={testSubjects} onSubmit={noop} />,
+    );
+
+    expect(
+      screen.getByRole('heading', {
+        name: 'Step 1 (current) Select a data set',
+      }),
+    ).toBeInTheDocument();
+
+    const radios = within(
+      screen.getByRole('group', {
+        name: 'Select a data set',
+      }),
+    ).getAllByRole('radio');
+    expect(radios).toHaveLength(2);
+    expect(radios[0]).toEqual(screen.getByRole('radio', { name: 'Subject 1' }));
+    expect(radios[1]).toEqual(screen.getByRole('radio', { name: 'Subject 2' }));
```

Comment:
For radios (and other form inputs), we can typically just use `getByLabel` which is faster to select. Would suggest refactoring to use these instead.

Would also do this for other radio instances in this test suite too.